x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index.tsx

```diff
@@ -306,3 +306,9 @@ const IndexParamsFields: React.FunctionComponent<ActionParamsProps<IndexActionPa
     </Fragment>
   );
 };
+
+// if the string == null or is empty, return null, else return string
+function nullableString(str: string | null | undefined) {
+  if (str == null || str.trim() === '') return null;
+  return str;
+}
```

Comment:
can e.target.value actually ever be null?
If not, this can be simplified.

Reply:
before this change, the field in question could only be `string | undefined` when coming from the server; after the change with the action type config schema, it can now be `string | null`.  Better question is, can it ever be `undefined` now :-).  There is some code you can see in the context ^^^ `if (executionTimeField === undefined)`, so I figured - there may still be a lurking case, or the target values may be initialized to undefined or such anyway.  It's also the case that this field may be passed as an `undefined` value (not passed at all, basically) to the server, which will then get treated as `null` there.

Reply:
For me is always hard to get why we should have 3 states like undefined, string and null. I know sometimes because of javascript... but it still seems to be weird to send null or not send the property at all works ok in both cases. 
With the new changes it seems to be always null. If you think it is reasonable, we can have `executionTimeField: string | null;` instead of `executionTimeField?: string | null;`, but I don't mind to have it as it is.

Reply:
Seems like we could change it to be just null con the client-side; the server-side code can handle either as input (create or update), but will never return `undefined` on output (eg, a get or list). 

I'll push another commit in a bit ...

Reply:
I checked what happens when you create a new connector, and when it gets into the code below, the config fields are all `undefined`.  As such, I think I'm going to leave that code the way it is, to handle both `undefined` and `null`.

https://github.com/elastic/kibana/blob/ad29b506b2fd8aadb93df5d97ae6d4a3c75538a2/x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index.tsx#L71-L77

I do have some tests to fix, so more commits incoming anyway :-)